Add subfeatures details

Akusem opened the pull request:
I have finally some time available to work on the issue #1234, this is basically the work I had done last time we had discussed about, already 2 weeks ago. 
It display subfeatures (like transcript, exon or CDS) information. To avoid slowing down the browser, only the transcripts and mRNA are displayed until the user click on `load additional subfeatures` .

Ex before clicking:
![image](https://user-images.githubusercontent.com/24433720/94944654-d8676080-04d9-11eb-8b94-fd96d12325f1.png)

After:
![image](https://user-images.githubusercontent.com/24433720/94949487-47948300-04e1-11eb-9bec-fabafbac3a03.png)

cmdcolin commented:
@Akusem thanks for submitting this

I tested it out and it's pretty good, especially as a first pass. It might be that it is acceptable as is then

Some random comments

1) I might not use by default `const subfeaturesOmit = ['gene_id', 'gene_name', 'gene_type', 'gene_status']`. Even this is duplicated across multiple transcript subfeatures it is maybe not the best default setting to have as people might want to see these fields
2) The subfeatures are technically probably children of a specific transcript but they are loaded below all the transcripts
screenshot from wormbase with the JBrowse 1 featuresequence plugin
3) I feel like, probably a really awesome ideal would be to de-emphasize subfeatures like cds, exon (which you have already done) and have something like this


![wormbase org_tools_genome_jbrowse-simple__data=data%2Fc_elegans_PRJNA13758 loc=II%3A159133 160002 tracks=Curated_Genes highlight=](https://user-images.githubusercontent.com/6511937/96634322-53a58080-12e8-11eb-9d07-58980d5605f6.png)

That is a screenshot of the featuresequence plugin in jbrowse 1 on wormbase

Of course that is a longer PR so we might be able to accept this as is, but I want to open our mind to the future of what a transcript feature detail panel *could be*

cmdcolin commented:
Random other comment: non gene features have an "empty subfeatures area" that is visible below their details

![localhost_3000__config=test_data%2Fvolvox%2Fconfig json](https://user-images.githubusercontent.com/6511937/96634962-3f15b800-12e9-11eb-90f2-d7b57d10db7f.png)

Akusem commented:
Hi @cmdcolin,
I've fixed the 'empty subfeatures area', removed `subfeaturesOmit`, and finally i've modified the omited field to allow subfeatures to display `start`, `end` and `name` field, who can be useful to the user I think ?

I have to say that the output of featuresequence plugin in jbrowse 1 look superbe, and I agree that it would be much nicer to have than a simple list of subfeatures like now ! 

Just two remark:
1. First, I don't think the `feature detail` panel is large enought for this without looking too compressed, so we will maybe have to enlarge it ?
2. Secondly, I'm not sure on how to access the sequence to display from BaseFeaturesDetails ? Maybe from `model` props ?

rbuels commented on packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx:
```diff
@@ -223,14 +223,107 @@ export const Attributes: FunctionComponent<AttributeProps> = props => {
+  let subfeaturesToRender = getSubfeaturesToRender(feature.subfeatures, [])
+  // If not loaded, display only transcript and mRNA
+  if (!subfeaturesLoaded) {
+    subfeaturesToRender = subfeaturesToRender.filter(subfeature =>
+      ['transcript', 'mRNA'].includes(subfeature.title),
+    )
```
This filtering assumes a specific structure of subfeatures, which is not appropriate for this completely-generic feature detail implementation. Could you remove it?

Akusem replied:
@rbuels If I understand correctly it should only display a `load subfeatures` button at start and then load every subfeatures ?
Example:
![image](https://user-images.githubusercontent.com/24433720/100219305-19965280-2f16-11eb-8427-ebda2bea1f29.png)

![image](https://user-images.githubusercontent.com/24433720/100219176-f9669380-2f15-11eb-895a-8d55768e280a.png)

rbuels replied:
What I meant is, this is specific to "transcript" and "mRNA" subfeatures. This feature track is used to display many things, not just genes. Filtering here for only mRNA and transcript subfeatures means that other types of features won't be able to display their subfeatures. I see you removed it in 9588d98 anyway, so you probably understand what I mean already. :-)

rbuels reviewed (commented):
sorry for the slow response @Akusem. we're still interested in merging this, I think it looks quite good in general. My main concern is that it is specifically for gene features (assumes transcript and mRNA feature types), but this feature detail implementation is meant to be very general and can't assume any specific structure of subfeatures.

cmdcolin commented:
@rbuels I made an alternative to this PR in https://github.com/GMOD/jbrowse-components/pull/1529

It is a little more straightforward to me, and includes nested subfeatures

cmdcolin commented:
We could probably merge this as is and then follow up with #1529

cmdcolin commented:
Thanks a bunch @Akusem for pushing on this issue, it was really great :)
